In [1]:
#importing libraries/dependencies

import pandas as pd

#for mysql 
import pymysql
from sqlalchemy import create_engine


In [2]:
#Reading csv and storing in dataframe df
df = pd.read_csv('Walmart.csv', encoding_errors='ignore')

In [3]:
#Checking data for nulls,duplicates,error values

#describing data to get statistics of a DataFrame ie,"Count" field provides the number of non-null values.
df.describe()

invoice_id      quantity        rating  profit_margin
count  10051.000000  10020.000000  10051.000000   10051.000000
mean    5025.741220      2.353493      5.825659       0.393791
std     2901.174372      1.602658      1.763991       0.090669
min        1.000000      1.000000      3.000000       0.180000
25%     2513.500000      1.000000      4.000000       0.330000
50%     5026.000000      2.000000      6.000000       0.330000
75%     7538.500000      3.000000      7.000000       0.480000
max    10000.000000     10.000000     10.000000       0.570000

The count for "Quantity" shows 10020 which shows it has null values compared to other columns.

In [4]:
#Checking count of null values
df.isnull().sum()

invoice_id         0
Branch             0
City               0
category           0
unit_price        31
quantity          31
date               0
time               0
payment_method     0
rating             0
profit_margin      0
dtype: int64

Unit_Price and Quantity has 31 null values, so we will remove these values

In [5]:
#droppping all rows with missing records/null values
df.dropna(inplace=True)


In [6]:
# verifying the updated values
df.isnull().sum()

invoice_id        0
Branch            0
City              0
category          0
unit_price        0
quantity          0
date              0
time              0
payment_method    0
rating            0
profit_margin     0
dtype: int64

No null values now in data, we still can have duplicate values.

In [7]:
# duplicates row
df.duplicated().sum()

np.int64(51)

51 dulicate rows so we will remove these

In [8]:
#Removing duplicate rows
df.drop_duplicates(inplace=True)
#verify updated rows
df.duplicated().sum()

np.int64(0)

Data is cleansed for duplicate value and null values.

In [9]:
#datatypes of columns
df.dtypes

invoice_id          int64
Branch             object
City               object
category           object
unit_price         object
quantity          float64
date               object
time               object
payment_method     object
rating            float64
profit_margin     float64
dtype: object

The datatype for unit_price is object which should be float for numeric calculations.

In [ ]:
#Coverting datatype of 'unit_price' column to float using astype function.
#Note:The column contains the symbol "$" we have to replace it before converting it to float as float doesnt take a character value.
df['unit_price']=df['unit_price'].str.replace('$','').astype(float)

In [13]:
#verify updated datatype
df.dtypes

invoice_id          int64
Branch             object
City               object
category           object
unit_price        float64
quantity          float64
date               object
time               object
payment_method     object
rating            float64
profit_margin     float64
dtype: object

Now we will add a new column 'Total' to calculae teh total sale

In [14]:
df['total'] = df['unit_price'] * df['quantity']

In [15]:
#displaying first 5 rows of data
df.head()

invoice_id   Branch         City                category  unit_price  \
0           1  WALM003  San Antonio       Health and beauty       74.69   
1           2  WALM048    Harlingen  Electronic accessories       15.28   
2           3  WALM067  Haltom City      Home and lifestyle       46.33   
3           4  WALM064      Bedford       Health and beauty       58.22   
4           5  WALM013       Irving       Sports and travel       86.31   

   quantity      date      time payment_method  rating  profit_margin   total  
0       7.0  05/01/19  13:08:00        Ewallet     9.1           0.48  522.83  
1       5.0  08/03/19  10:29:00           Cash     9.6           0.48   76.40  
2       7.0  03/03/19  13:23:00    Credit card     7.4           0.33  324.31  
3       8.0  27/01/19  20:33:00        Ewallet     8.4           0.33  465.76  
4       7.0  08/02/19  10:37:00        Ewallet     5.3           0.48  604.17

Now we will connect to Mysql database using create_engine() of sqlalchemy library

In [16]:
#Connecting to mysql server
hostname='127.0.0.1'
password='1234567890'
username='root'
port=3306
database='walmart'
engine=create_engine('mysql+pymysql://'+username+':'+password+'@'+hostname+':'+str(port)+'/'+database)
try:
    engine
    print("connnection successfull")

except:
    print("error")


connnection successfull


Now we will write records stored in a DataFrame to a SQL database.

In [17]:
df.to_sql(name='walmart',con=engine,if_exists='append',index=False)  

9969

This will create a sql table named "Walmart" in the sql server which you can further anlayze using mysql workbench